agent/src/main.rs

```diff
@@ -57,6 +55,22 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>
         run_metrics_server().await.unwrap();
     }));
 
+    let discovery_handler_map = Arc::new(Mutex::new(HashMap::new()));
+    let discovery_handler_map_clone = discovery_handler_map.clone();
+    let (new_discovery_handler_sender, _): (
+        broadcast::Sender<String>,
+        broadcast::Receiver<String>,
+    ) = broadcast::channel(4);
```

Comment:
channel is used for multi-consumer/multi-producer scenarios. Is [mpsc](https://docs.rs/tokio/0.2.9/tokio/sync/mpsc/index.html) more appropriate here?

Reply:
I am using broadcast any time I need multiple consumers. In this scenario, the registration service sends a message to all running `DiscoveryOperators` any time a new DiscoveryHandler registers. Each `DiscoveryOperator` subscribes to the `new_discovery_handler_sender`. This scenario fits [watch](https://docs.rs/tokio/0.2.9/tokio/sync/watch/index.html) better than [broadcast](https://docs.rs/tokio/0.2.9/tokio/sync/broadcast/index.html) but i worry about watch only retaining the latest sent value. `broadcast` doesnt drop something from a channel until all receivers have received it.

Reply:
added in fa4eba408f57f22a